library/Booster/Pattern/Util.hs

```diff
@@ -199,18 +202,29 @@ filterTermSymbols check = cata $ \case
         let concatSym = concatSymbol $ KListMeta def
             elemSym = klistElementSymbol def
             unitSym = unitSymbol $ KListMeta def
-         in if null heads
-                then [unitSym | check unitSym]
-                else filter check [concatSym, elemSym] <> concat heads
+         in case heads of
+                [] -> [unitSym | check unitSym]
+                [x] -> [elemSym | check elemSym] <> x
+                more -> filter check [concatSym, elemSym] <> concat more
```

Comment:
we don't really need to check any of the symbols in list since we know they are always total.

Reply:
Good point. True for the currently single use of `filterTermSymbols`. In general, `check`  can be anything, though, so I would prefer to leave as-is.